reconcile/external_resources/manager.py

```diff
@@ -311,7 +313,9 @@ def _build_external_resource(
 
     def _serialize_resource_input(self, resource: ExternalResource) -> str:
         return json.dumps(
-            resource.dict(exclude={"data": {FLAG_RESOURCE_MANAGED_BY_ERV2}})
+            resource.dict(
+                exclude={"data": {FLAG_RESOURCE_MANAGED_BY_ERV2, FLAG_DELETE_RESOURCE}}
+            )
```

Comment:
why do we need to exclude those fields? There is a similar method `serialize_input` in `ExternalResource`, and to serialize json, can directly call `resource.json(` instead of `resource.dict(` then `json.dumps`

Reply:
- These fields are not expected in the module input. Since the `overrides` parameter doesn't have a schema, all the overridden parameters are passed to the terraform object directly (the same way as terraform-resources). So all parameters not expected in the input shall be removed. 

- I changed the call to `resource.json()` and removed the method in ExternalResource (this was used in an old implementation when the INPUT was passed as an env var)

Reply:
Is there a way to ignore unexpected input or define a clear mapping? Otherwise we have to remember edit this part whenever we put new fields in `data`